lib/parsers/webp_parser.rb

```diff
@@ -0,0 +1,159 @@
+    # integers (minus one). The 4 remaining bits consist of a 1-bit flag indicating whether alpha is used, and a 3-bit
+    # version that is always zero.
+    dimensions = read_little_endian_int_32
+    width = (dimensions & 0x3fff) + 1
+    height = (dimensions >> 14 & 0x3fff) + 1
+    has_transparency = (dimensions >> 28 & 0x1) == 1
+
+    create_image(width, height, has_transparency: has_transparency)
+  end
+
+  def read_extended_data
+    # After the common RIFF header bytes, the extended file format has a series of 1-bit flags to signify the presence
+    # of optional information. These flags are as follows:
+    #
+    # |0|1|2|3|4|5|6|7|
+    # +-+-+-+-+-+-+-+-+
+    # |Rsv|I|L|E|X|A|R|
+    #
+    # Where:
+    #   - Rsv & R = Reserved - Should be 0.
+    #   - I = Set if file contains an ICC profile.
+    #   - L = Set if file contains transparency information.
+    #   - E = Set if file contains Exif metadata.
+    #   - X = Set if file contains XMP metadata.
+    #   - A = Set if file is an animated image.
+    flags = read_int_8
+    has_transparency = flags & 0x10 != 0
+    has_exif_metadata = flags & 0x08 != 0
+    has_xmp_metadata = flags & 0x04 != 0
+    has_multiple_frames = flags & 0x02 != 0
+
+    # The flags are followed by three reserved bytes of zeros, and then by the width and height, respectively - each
+    # occupying three bytes and each one less than the actual canvas measurements.
+    safe_skip(@buf, 3)
+    dimensions = safe_read(@buf, 6).unpack('VS')
+    width = (dimensions[0] & 0xffffff) + 1
+    height = (dimensions[0] >> 24 | dimensions[1] << 8 & 0xffffff) + 1
+
+    image = create_image(width, height, has_multiple_frames: has_multiple_frames, has_transparency: has_transparency)
+    augment_image(image) if has_exif_metadata || has_xmp_metadata || has_multiple_frames
+    image
+  end
+
+  def create_image(width, height, has_multiple_frames: false, has_transparency: false)
+    FormatParser::Image.new(
+      content_type: WEBP_MIME_TYPE,
+      format: :webp,
+      has_multiple_frames: has_multiple_frames,
+      has_transparency: has_transparency,
+      height_px: height,
+      width_px: width
+    )
+  end
+
+  def augment_image(image)
+    # We're going to scan the file looking for the EXIF, XMP and/or ANMF chunks.
+    exif = nil
+    xmp = nil
+    num_frames = 0
+    loop do
+      fourcc, chunk_size = safe_read(@buf, 8).unpack('A4V')
+      safe_skip(@buf, 1) if chunk_size.odd? # Padding byte of 0 added if chunk size is odd.
+      case fourcc
+      when 'EXIF'
+        exif ||= exif_from_tiff_io(StringIO.new(safe_read(@buf, chunk_size)))
+      when 'XMP'
+        xmp = Nokogiri::XML(safe_read(@buf, chunk_size))
+      when 'ANMF'
+        num_frames += 1 if image.has_multiple_frames
+        safe_skip(@buf, chunk_size)
+      else
+        safe_skip(@buf, chunk_size)
+      end
+    end
+  rescue FormatParser::IOUtils::InvalidRead
+    intrinsics = {}
+    if exif
```

Comment:
Looking at the JPG parser, it seems that EXIF data is always set in the intrinsics, even when it's empty. Maybe we can do the same, for consistency reasons.